### PR TITLE
Noise2inverse and MSD_Net

### PR DESCRIPTION
**Pull request by cshoebridge**
LION Compliant implementation of MSD_Net aswell as benchmarking script
Noise2Inverse reimplementation to align with paper
Various refactors to improve type safety, robustness and extendability of LIONSolver and LIONModel

### CODE REVIEW

**Comment by cshoebridge** [comment]
Pretty much just consists of the things we discussed yesterday.
Still lots of clean up to do, especially with some of the throwaway scripts, but thought I'd keep them in for now until proper demos get made.

File: LION/classical_algorithms/sirt.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import torch
+from LION.CTtools.ct_geometry import Geometry
+from LION.CTtools.ct_utils import make_operator
+from ts_algorithms import sirt as ts_sirt
+import tomosipo as ts
+
+from LION.exceptions.exceptions import NoDataException
+
+
+def sirt_from_geo(
```
Comment (AnderBiguri) [inline LION/classical_algorithms/sirt.py:11]: Maybe interesting to discuss: 

One of the advantages of not doing static typing is that we can have 1 SIRT only. Classically in satic typing lagnuages you can overload functions, but the advantage of dynamically typed ones like python is that instead of having something like `sirt_from_geo` we can simply do:
```
assert isinstance(geo,Geometry) or isinstance(geo,ts.Operator.Operator), seconed input must be a ct.Geometry or an operator. 
if isinstance(geo,Geometry):
    op=make_operator(geo)
```

opinions?

File: scripts/benchmarking/msdnet_benchmarks.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
+#%% This example shows how to train MSDNet for full angle, noisy measurements.
```
Comment (AnderBiguri) [inline scripts/benchmarking/msdnet_benchmarks.py:1]: This folder is for a paper we did on bechmarking. Bad name, but I haven't gotten around to fix it yet. This should be in a different folder, but not sure where yet.

File: LION/classical_algorithms/sirt.py (hunk 1 above)
Comment (cshoebridge) [reply LION/classical_algorithms/sirt.py:11]: also works! I don't really prefer one over the other, to be honest, I suppose it's a question of consistency amongst methods.

Comment (cshoebridge) [reply LION/classical_algorithms/sirt.py:11]: one other thing I'd say is if we do something like that, it's adding an extra check every time where we may not need it. From memory isinstance uses reflection , which is quite slow...

File: scripts/benchmarking/msdnet_benchmarks.py (hunk 2 above)
Comment (cshoebridge) [reply scripts/benchmarking/msdnet_benchmarks.py]: added some extra folders, probably worth having a clean through example_scripts at some point. It's become just a bit of a dump of whatever doesn't fit anywhere else (atleast that's how I've been using it, oops). Suppose that's for later down the line when proper demos get made though.

File: LION/experiments/ct_experiments.py [hunk 3]
```diff
@@ -12,6 +12,8 @@
 import warnings
 from abc import ABC, abstractmethod, ABCMeta
 
+from zmq import device
```
Comment (AnderBiguri) [inline LION/experiments/ct_experiments.py:15]: remove

File: LION/utils/activations.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import torch.nn as nn
+
+ACTIVATIONS = {"ReLU": nn.ReLU, "sigmoid": nn.Sigmoid}
```
Comment (AnderBiguri) [inline LION/utils/activations.py:3]: Lets not do this (for now)
